java/arcs/core/allocator/Arc.kt

```diff
@@ -147,8 +147,15 @@ class Arc internal constructor(
         if (arcState == state) return this
 
         val deferred: CompletableDeferred<Arc> = CompletableDeferred()
-        onArcStateChangeFiltered(state) {
-            deferred.complete(this@Arc)
+
+        onArcStateChange {
+            when(it) {
+                Running -> deferred.complete(this@Arc)
+                Error -> deferred.completeExceptionally(
+                    Exception("Arc failed to start.")
```

Comment:
Create an `AllocatorException`, or, are module/purpose-specific exceptions less the pattern for Kt?

Reply:
Yes, good call; I didn't intend to leave this as the generic `Exception` type.

Reply:
I introduced an `ArcErrorException` type, this can be throw by waitFor, and perhaps also used by the eventualy `onError` listener that @cromwellian mentioned.  Too specific?